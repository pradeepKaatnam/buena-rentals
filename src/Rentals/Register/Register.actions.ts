import { IFinancialInformation, IFormField, IPersonalInformation } from "./Register.models";
import { IRegisterService } from "./Register.service";

export enum REGISTER_ACTIONS {
  UPDATE_PERSONAL_INFO_ENTRY = 'UPDATE_PERSONAL_INFO_ENTRY',
  UPDATE_FINANCIAL_INFO_ENTRY = 'UPDATE_FINANCIAL_INFO_ENTRY', 
  UPDATE_PERSONAL_INFO = 'UPDATE_PERSONAL_INFO',
  UPDATE_FINANCIAL_INFO = 'UPDATE_FINANCIAL_INFO',  
  UPDATE_API_FLOW = 'UPDATE_API_FLOW',
}

export type IRegisterDispatchActions = | {
    type: REGISTER_ACTIONS.UPDATE_PERSONAL_INFO_ENTRY;
    data: { key: string, value: IFormField };
  }
| {
    type: REGISTER_ACTIONS.UPDATE_FINANCIAL_INFO_ENTRY;
    data: { key: string, value: IFormField };
  }
| {
    type: REGISTER_ACTIONS.UPDATE_FINANCIAL_INFO;
    data: IFinancialInformation;
}
| {
    type: REGISTER_ACTIONS.UPDATE_PERSONAL_INFO;
    data: IPersonalInformation;
}
| {
    type: REGISTER_ACTIONS.UPDATE_API_FLOW;
    data: {loading?: boolean;
        errorMessage?: string;};
}

export interface IRegisterActions {
    updatePersonalInformationEntry:(key: string, value: IFormField) => void;
    updateFinancialInformationEntry:(key: string, value: IFormField) => void;
    updatePersonalInformation:(data: IPersonalInformation) => void;
    updateFinancialInformation:(data: IFinancialInformation) => void;
    updateAPIFlowState: (loading: boolean) => void;
    register: (personalInfo: IPersonalInformation, financialInformation: IFinancialInformation) => Promise<void>;
}

export class RegisterActions implements IRegisterActions {
    private readonly dispatch: React.Dispatch<IRegisterDispatchActions>;
    private readonly service: IRegisterService;

    constructor(dispatch: React.Dispatch<IRegisterDispatchActions>, service: IRegisterService) {
        this.dispatch = dispatch;
        this.service = service;
    }

    updateAPIFlowState(loading: boolean) {
        this.dispatch({
            type: REGISTER_ACTIONS.UPDATE_API_FLOW,
            data: {loading: loading},
        });
    };

    async register(personalInfo: IPersonalInformation, financialInformation: IFinancialInformation): Promise<void> {
        this.dispatch({
            type: REGISTER_ACTIONS.UPDATE_API_FLOW,
            data: { loading: true },
        });
        const data = await this.service.register(personalInfo, financialInformation);
        this.dispatch({
            type: REGISTER_ACTIONS.UPDATE_API_FLOW,
            data: { loading: false, errorMessage: data?.errors && data.errors.length > 0 ? data.errors[0] : '' },   
        });
    }

    updatePersonalInformationEntry(key: string, value: IFormField) {
        this.dispatch({
            type: REGISTER_ACTIONS.UPDATE_PERSONAL_INFO_ENTRY,
            data: {
                key: key,
                value: value,
            },
        });
    }

    updateFinancialInformationEntry(key: string, value: IFormField) {
        this.dispatch({
            type: REGISTER_ACTIONS.UPDATE_FINANCIAL_INFO_ENTRY,
            data: {
                key: key,
                value: value,
            },
        });
    }

    updatePersonalInformation(data: IPersonalInformation) {
        this.dispatch({
            type: REGISTER_ACTIONS.UPDATE_PERSONAL_INFO,
            data,
        });
    }

    updateFinancialInformation(data: IFinancialInformation) {
        this.dispatch({
            type: REGISTER_ACTIONS.UPDATE_FINANCIAL_INFO,
            data,
        });
    }
}
