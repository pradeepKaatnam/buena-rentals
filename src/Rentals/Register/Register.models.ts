export interface IRegisterProps {

}

export interface IPersonalInformationProps {
    formData: IPersonalInformation;
    onFormChange: (formType: string, key: string, value: string) => void;
}

export interface IFinancialInformationProps {
    formData: IFinancialInformation;
    onFormChange: (formType: string, key: string, value: string) => void;
}   

export interface IReviewProps {
    personalInformation: IPersonalInformation;
    financialInformation: IFinancialInformation;
}

export interface IPersonalInformation {
    firstName: string;
    lastName: string;
    addressLine1: string;
    addressLine2?: string;
    email: string;
    nationality: string;
}

export interface IFinancialInformation {
    employmentStatus: string;
    employer: string;
    occupation: string;
    workDuration: string;
    salaryIndication: number;
}

export interface IRegisterState {
    personalInformation: IPersonalInformation;
    financialInformation: IFinancialInformation;
    isSubmitting: boolean;
    isSubmitted: boolean;
    errorMessage: string;
}
