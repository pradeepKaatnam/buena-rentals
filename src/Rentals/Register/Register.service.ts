export interface IRegisterService {
}
  
export class RegisterService implements IRegisterService {
}