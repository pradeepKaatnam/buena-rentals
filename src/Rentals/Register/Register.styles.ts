import { Theme } from '@mui/material';
import { createStyles, makeStyles } from '@mui/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    formLabel: {
      marginBottom: '8px',
      alignSelf: 'start',
    },
    textField: {
      marginBottom: '16px', // Add some styles for textField
    },
  })
);

export const RegisterStyles = {
    gridLeftContainer: {
        display: { xs: 'none', md: 'flex' },
        flexDirection: 'column',
        borderRight: { sm: 'none', md: '1px solid' },
        borderColor: { sm: 'none', md: 'divider' },
        alignItems: 'start',
        pt: 16,
        px: 10,
        gap: 4,
        backgroundColor: 'black',
      color: 'white',
    },
    rightGridContainer: {
        display: 'flex',
        flexDirection: 'column',
        maxWidth: '100%',
        width: '100%',
        backgroundColor: { xs: 'transparent', sm: 'background.default' },
        alignItems: 'start',
        pt: { xs: 6, sm: 16 },
        px: { xs: 2, sm: 10 },
        gap: { xs: 4, md: 8 },
    },
    rightGridBox: {
        display: 'flex',
        justifyContent: { sm: 'space-between', md: 'flex-end' },
        alignItems: 'center',
        width: '100%',
        maxWidth: { sm: '100%', md: 600 },
    },
    stepperBox: {
        display: { xs: 'none', md: 'flex' },
        flexDirection: 'column',
        justifyContent: 'space-between',
        alignItems: 'flex-end',
        flexGrow: 1,
    },
    steps: { 
        ':first-child': { pl: 0 }, 
        ':last-child': { pr: 0 },
        '& .MuiStepLabel-root .Mui-completed': {
            color: '#333333',  // Change color of completed steps
        },
        '& .MuiStepLabel-root .Mui-active': {
            color: '#000000', // Change color of active step
        },
        '& .MuiStepLabel-root .MuiStepLabel-label': {
            color: '#666666', // Change color of step labels
        },
    },
    stepperBoxMobile: {
      display: 'flex',
      flexDirection: 'column',
      flexGrow: 1,
      width: '100%',
      maxWidth: { sm: '100%', md: 600 },
      maxHeight: '720px',
      gap: { xs: 5, md: 'none' },
    },
    buttonsBox: {
      display: 'flex',
      flexDirection: { xs: 'column-reverse', sm: 'row' },
      alignItems: 'end',
      flexGrow: 1,
      gap: 1,
      pb: { xs: 12, sm: 0 },
      mt: { xs: 2, sm: 0 },
      mb: '60px',
    },
    loader: {
      position: 'fixed',
      top: 0,
      left: 0,
      width: '100%',
      height: '100%',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: 'rgba(255, 255, 255, 0.8)',
      zIndex: 9999,
  }
}

export default useStyles;