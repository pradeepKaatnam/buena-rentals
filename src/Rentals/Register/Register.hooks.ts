import { useEffect, useReducer } from 'react';
import { IRegisterState } from './Register.models';
import { IRegisterActions, RegisterActions } from './Register.actions';
import { registerReducer } from './Register.reducer';
import { IRegisterService, RegisterService } from './Register.service';

export const useInit = (
): {
  state: IRegisterState;
  actions: IRegisterActions;
} => {
  const [state, dispatch] = useReducer(registerReducer, {
    financialInformation: {},
    personalInformation: {},
  } as IRegisterState);
  
  const service: IRegisterService = new RegisterService();
  const actions: IRegisterActions = new RegisterActions(dispatch, service);
  useEffect(() => {}, []);
  return { state, actions };
};
