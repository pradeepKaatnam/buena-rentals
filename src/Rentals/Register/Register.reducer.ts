import { IRegisterDispatchActions, REGISTER_ACTIONS } from "./Register.actions";
import { IRegisterState } from "./Register.models";

export const registerReducer = (state: IRegisterState, action: IRegisterDispatchActions): IRegisterState => {
    switch (action.type) {
        case REGISTER_ACTIONS.UPDATE_PERSONAL_INFO_ENTRY: {
          return {
            ...state,
            personalInformation: {
              ...state.personalInformation,
              [action.data.key]: action.data.value
            },
          };
        }
        case REGISTER_ACTIONS.UPDATE_FINANCIAL_INFO_ENTRY: {
          return {
            ...state,
            financialInformation: {
              ...state.financialInformation,
              [action.data.key]: action.data.value,
            },
          };
        }
        case REGISTER_ACTIONS.UPDATE_FINANCIAL_INFO: {
            return { ...state,  financialInformation: action.data };
        }
        case REGISTER_ACTIONS.UPDATE_PERSONAL_INFO: {
            return { ...state,  personalInformation: action.data };
        }
        case REGISTER_ACTIONS.UPDATE_API_FLOW: {
            return { ...state, ...action.data };
        }
        default: {
          throw new Error(`Unhandled action type`);
        }
      }
};
  