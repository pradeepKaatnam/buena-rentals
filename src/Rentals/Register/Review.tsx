import { Divider, Stack, Typography } from '@mui/material';
import React from 'react';
import Grid from '@mui/material/Grid2';
import { IReviewProps } from './Register.models';

const Review: React.FC<IReviewProps> = (props: IReviewProps) => {
    const employmentStatusMap: { [key: number]: string } = {
        1: 'Employed',
        2: 'Self-employed',
        3: 'Student',
        4: 'Other',
      };

    return (
        <Stack
        direction="column"
        divider={<Divider flexItem />}
        spacing={2}
        sx={{ my: 2 }}
      >
        <div>
          <Typography variant="subtitle2" gutterBottom sx={{
                fontWeight: 'bold',
                marginBottom: 2,
                color: 'black',
            }}>
            Personal Information
          </Typography>
          <Typography gutterBottom>{props.personalInformation.firstName.value} {props.personalInformation.lastName.value}</Typography>
          <Typography gutterBottom sx={{ color: 'text.secondary' }}>
            {props.personalInformation.addressLine1.value}
            {props.personalInformation.addressLine2.value && `, ${props.personalInformation.addressLine2.value}`}
          </Typography>
          <Typography gutterBottom>{props.personalInformation.email.value}</Typography>
          <Typography gutterBottom>{props.personalInformation.nationality.value}</Typography>
        </div>
        <div>
          <Typography variant="subtitle2" gutterBottom sx={{
                fontWeight: 'bold',
                marginBottom: 2,
                color: 'black',
            }}>
            Financial Information
          </Typography>
          <Grid container>
            <Stack
                direction="row"
                spacing={1}
                useFlexGap
                sx={{ width: '100%', mb: 1 }}
                >
                    <Typography variant="body1" sx={{ color: 'text.secondary' }}>
                     Employment Status:
                    </Typography>
                    <Typography variant="body2" style={{ marginTop: '2px' }}>
                    {employmentStatusMap[Number(props.financialInformation.employmentStatus.value)]}
                    </Typography>
            </Stack>
            <Stack
                direction="row"
                spacing={1}
                useFlexGap
                sx={{ width: '100%', mb: 1 }}
                >
                    <Typography variant="body1" sx={{ color: 'text.secondary' }}>
                     Occupation:
                    </Typography>
                    <Typography variant="body2" style={{ marginTop: '2px' }}>
                    {props.financialInformation.occupation.value}
                    </Typography>
            </Stack>
            <Stack
                direction="row"
                spacing={1}
                useFlexGap
                sx={{ width: '100%', mb: 1 }}
                >
                    <Typography variant="body1" sx={{ color: 'text.secondary' }}>
                     Employer:
                    </Typography>
                    <Typography variant="body2" style={{ marginTop: '2px' }}>
                    {props.financialInformation.employer.value}
                    </Typography>
            </Stack>
            <Stack
                direction="row"
                spacing={1}
                useFlexGap
                sx={{ width: '100%', mb: 1 }}
                >
                    <Typography variant="body1" sx={{ color: 'text.secondary' }}>
                     Work Duration:
                    </Typography>
                    <Typography variant="body2" style={{ marginTop: '2px' }}>
                    {props.financialInformation.workDuration.value}
                    </Typography>
            </Stack>
            <Stack
                direction="row"
                spacing={1}
                useFlexGap
                sx={{ width: '100%', mb: 1 }}
                >
                    <Typography variant="body1" sx={{ color: 'text.secondary' }}>
                     Salary Indication:
                    </Typography>
                    <Typography variant="body2" style={{ marginTop: '2px' }}>
                    {props.financialInformation.salaryIndication.value}
                    </Typography>
            </Stack>
          </Grid>
        </div>
      </Stack>
    );
};

export default Review;