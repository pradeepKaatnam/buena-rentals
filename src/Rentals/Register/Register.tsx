import React from 'react';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import Grid from '@mui/material/Grid2';
import Stack from '@mui/material/Stack';
import Step from '@mui/material/Step';
import StepLabel from '@mui/material/StepLabel';
import Stepper from '@mui/material/Stepper';
import Typography from '@mui/material/Typography';
import RentalsInfo from './RentalsInfo';
import PersonalInformation from './Forms/PersonalInformation';
import FinancialInformation from './Forms/FinancialInformation';
import Review from './Review';
import useStyles, { RegisterStyles } from './Register.styles';
import ChevronRightRoundedIcon from '@mui/icons-material/ChevronRightRounded';
import ChevronLeftRoundedIcon from '@mui/icons-material/ChevronLeftRounded';
import { useInit } from './Register.hooks';
import { validateFinancialInformation, validatePersonalInformation } from '../../common/utilities/form-utilities';
import { CircularProgress } from '@mui/material';

const Register: React.FC = () => {
    const steps = ['Personal Info', 'Financial Info', 'Review'];
    const [activeStep, setActiveStep] = React.useState(0);
    const { state, actions } = useInit();
    const classes = useStyles();

    const getStepContent = (step: number) => {
        switch (step) {
            case 0:
                return <PersonalInformation formData={state.personalInformation} onFormChange={onFormChange} />;
            case 1:
                return <FinancialInformation formData={state.financialInformation} onFormChange={onFormChange} />;
            case 2:
                return <Review personalInformation={state.personalInformation} financialInformation={state.financialInformation} />;
            default:
                return 'Unknown step';
        }
    }

    const handleBack = () => {
        setActiveStep((prevActiveStep) => prevActiveStep - 1);
    }

    const handleNext = async () => {
        if(activeStep === 0) {
            const isValidForm = validatePersonalInformation(state.personalInformation);
            if(!isValidForm) {
                actions.updatePersonalInformation(state.personalInformation);
                return;
            }
        } else if (activeStep === 1) {
            const isValidForm = validateFinancialInformation(state.financialInformation);
            if(!isValidForm) {
                actions.updateFinancialInformation(state.financialInformation);
                return;
            }
        }

        if(activeStep === 2) {
            await actions.register(state.personalInformation, state.financialInformation);
        }

        setActiveStep((prevActiveStep) => prevActiveStep + 1);
    }

    const onFormChange = (formType: string, key: string, value: string) => {    
        var formField = {
            value: value,
            error: value === '' ? 'This field is required' : ''
        };
        if(formType === 'personalInformation') {
            actions.updatePersonalInformationEntry(key, formField);
        } else {
            actions.updateFinancialInformationEntry(key, formField);
        }
    }

    return <>
        {state.loading && (
            <Box sx={RegisterStyles.loader}>
                <CircularProgress />
            </Box>
        )}
        <Grid container sx={{ height: { xs: '100%', sm: '100dvh' } }}>
          <Grid
            size={{ xs: 12, sm: 5, lg: 4 }}
            sx={RegisterStyles.gridLeftContainer}
          >
            <RentalsInfo/>
          </Grid>
          <Grid
            size={{ sm: 12, md: 7, lg: 8 }}
            sx={RegisterStyles.rightGridContainer}
          >
            <Box
              sx={RegisterStyles.rightGridBox}
            >
              <Box
                sx={RegisterStyles.stepperBox}
              >
                <Stepper
                  id="desktop-stepper"
                  activeStep={activeStep}
                  sx={{ width: '100%', height: 40 }}
                >
                  {steps.map((label) => (
                    <Step
                      sx={RegisterStyles.steps}
                      key={label}
                    >
                      <StepLabel>{label}</StepLabel>
                    </Step>
                  ))}
                </Stepper>
              </Box>
            </Box>
            <Card sx={{ display: { xs: 'flex', md: 'none' }, width: '100%' }}>
              <CardContent
                sx={{
                  backgroundColor: 'black',
                  color: 'white',
                }}
              >
                <RentalsInfo/>
              </CardContent>
            </Card>
            <Box
              sx={RegisterStyles.stepperBoxMobile}
            >
              <Stepper
                id="mobile-stepper"
                activeStep={activeStep}
                alternativeLabel
                sx={{ display: { sm: 'flex', md: 'none' } }}
              >
                {steps.map((label) => (
                  <Step
                    sx={RegisterStyles.steps}
                    key={label}
                  >
                    <StepLabel
                      sx={{ '.MuiStepLabel-labelContainer': { maxWidth: '70px' } }}
                    >
                      {label}
                    </StepLabel>
                  </Step>
                ))}
              </Stepper>
              {activeStep === steps.length ? (
                <Stack spacing={2} useFlexGap>
                  <Typography variant="h1">ðŸŽ‰</Typography>
                  <Typography variant="h5">Thank you for registering!</Typography>
                  <br/>
                  <Button
                    variant="contained"
                    sx={{ alignSelf: 'start', width: { xs: '100%', sm: 'auto' }, backgroundColor: 'black',}}
                  >
                    Go to rentals
                  </Button>
                </Stack>
              ) : (
                <React.Fragment>
                  {getStepContent(activeStep)}
                  <Box
                    sx={[
                      RegisterStyles.buttonsBox,
                      activeStep !== 0
                        ? { justifyContent: 'space-between' }
                        : { justifyContent: 'flex-end' },
                    ]}
                  >
                    {activeStep !== 0 && (
                      <Button
                        startIcon={<ChevronLeftRoundedIcon />}
                        onClick={handleBack}
                        variant="text"
                        sx={{ display: { xs: 'none', sm: 'flex' } }}
                        style={{ color: 'black' }}
                      >
                        Previous
                      </Button>
                    )}
                    <Button
                      variant="contained"
                      endIcon={activeStep < steps.length - 1 && <ChevronRightRoundedIcon /> }
                      onClick={handleNext}
                      sx={{ width: { xs: '100%', sm: 'fit-content', backgroundColor: 'black',
                        color: 'white' } }}
                    >
                      {activeStep === steps.length - 1 ? 'Submit' : 'Next'}
                    </Button>
                  </Box>
                </React.Fragment>
              )}
            </Box>
          </Grid>
        </Grid>
    </>;
};

export default Register;