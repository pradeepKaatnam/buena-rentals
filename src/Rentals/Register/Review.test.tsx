import React from 'react';
import { render } from '@testing-library/react';
import Review from './Review';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import { IReviewProps } from './Register.models';

// Mock the useStyles module if necessary
jest.mock('./Register.styles', () => ({
  __esModule: true,
  default: () => ({
    root: 'root',
    // other mocked class names
  }),
}));

describe('Review Component', () => {
  const theme = createTheme();

  const mockProps: IReviewProps = {
    personalInformation: {
      firstName: { value: 'Pradeep', error: '' },
      lastName: { value: 'Kaatnam', error: '' },
      addressLine1: { value: '123 Main St', error: '' },
      addressLine2: { value: 'Apt 4B', error: '' },
      email: { value: 'pradeep.kaats@example.com', error: '' },
      nationality: { value: 'Indian', error: '' },
    },
    financialInformation: {
      employmentStatus: { value: '1', error: '' },
      occupation: { value: 'Software Engineer', error: '' },
      employer: { value: 'Microsoft', error: '' },
      workDuration: { value: '2 years', error: '' },
      salaryIndication: { value: '3.000 - 4.000', error: '' },
    },
  };

  it('matches the snapshot', () => {
    const { asFragment } = render(
        <Review {...mockProps} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});