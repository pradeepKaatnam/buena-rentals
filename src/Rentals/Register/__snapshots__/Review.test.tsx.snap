// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-14uew2-MuiStack-root"
  >
    <div>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-kiph5z-MuiTypography-root"
      >
        Personal Information
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1p5knw2-MuiTypography-root"
      >
        Pradeep Kaatnam
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-xyih9o-MuiTypography-root"
      >
        123 Main St, Apt 4B
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1p5knw2-MuiTypography-root"
      >
        pradeep.kaats@example.com
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1p5knw2-MuiTypography-root"
      >
        Indian
      </p>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-1quhdjb-MuiDivider-root"
    />
    <div>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-kiph5z-MuiTypography-root"
      >
        Financial Information
      </h6>
      <div
        class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-16epkjp-MuiGrid2-root"
      >
        <div
          class="MuiStack-root css-470g19-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-h6r3vf-MuiTypography-root"
          >
            Employment Status:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
            style="margin-top: 2px;"
          >
            Employed
          </p>
        </div>
        <div
          class="MuiStack-root css-470g19-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-h6r3vf-MuiTypography-root"
          >
            Occupation:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
            style="margin-top: 2px;"
          >
            Software Engineer
          </p>
        </div>
        <div
          class="MuiStack-root css-470g19-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-h6r3vf-MuiTypography-root"
          >
            Employer:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
            style="margin-top: 2px;"
          >
            Microsoft
          </p>
        </div>
        <div
          class="MuiStack-root css-470g19-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-h6r3vf-MuiTypography-root"
          >
            Work Duration:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
            style="margin-top: 2px;"
          >
            2 years
          </p>
        </div>
        <div
          class="MuiStack-root css-470g19-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-h6r3vf-MuiTypography-root"
          >
            Salary Indication:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
            style="margin-top: 2px;"
          >
            3.000 - 4.000
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
